### PR TITLE
MBS-6646: Compatibility with format_tiles sub-tiles

### PR DESCRIPTION
**Pull request by sh-csg**

### CODE REVIEW

File: lib.php [hunk 1]
```diff
@@ -196,6 +196,7 @@ function learningmap_cm_info_view(cm_info $cm) : void {
         $cm->set_extra_classes('label'); // ToDo: Add extra CSS.
         $PAGE->requires->js_call_amd('mod_learningmap/manual-completion-watch', 'init',
             ['coursemodules' => learningmap_get_place_cm($cm)]);
+        $cm->set_custom_cmlist_item(true);
```
Comment (PhMemmel) [inline lib.php:199]: This is likely to break compatibility for moodle 3.9 to 3.11 as this function has been introduced in moodle 4

**Comment by sh-csg** [comment]
(to avoid creating a separate moodle 4 version)

**Comment by PhMemmel** [comment]
This fixes the issue for format_tiles (and maybe other issues with course formats as well which rely on the newly introduced api function). If the learningmap is configured to show on the course page it will now show, no matter if format_tiles displays activites as subtiles or not. If the learningmap is configured to have its own view page it is shown as subtile as expected. Clicking on the subtile will open the separate view page.

**Comment by PhMemmel** [comment]
This also fixes a small debug notification error (empty check in `mapworker`) which is not related to the format_tiles fix

**Pull request merged**